pkg/agent/attestor/attestor_test.go

```diff
@@ -0,0 +1,169 @@
+
+	"crypto/x509"
+	"github.com/spiffe/spire/proto/agent/keymanager"
+	"github.com/spiffe/spire/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/proto/api/node"
+	"github.com/spiffe/spire/proto/common"
+	"github.com/spiffe/spire/test/mock/agent/catalog"
+	"github.com/spiffe/spire/test/mock/proto/agent/keymanager"
+	"github.com/spiffe/spire/test/mock/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/test/mock/proto/api/node"
+	"github.com/spiffe/spire/test/util"
+	"github.com/stretchr/testify/suite"
+	"path"
+	"testing"
+)
+
+type AttestorTestSuite struct {
+	suite.Suite
+
+	ctrl *gomock.Controller
+
+	attestor     Attestor
+	catalog      *mock_catalog.MockCatalog
+	nodeAttestor *mock_nodeattestor.MockNodeAttestor
+	keyManager   *mock_keymanager.MockKeyManager
+	nodeClient   *mock_node.MockNodeClient
+	config       *Config
+	expectation  *node.SvidUpdate
+}
+
+func (s *AttestorTestSuite) SetupTest() {
+	s.ctrl = gomock.NewController(s.T())
+
+	s.nodeAttestor = mock_nodeattestor.NewMockNodeAttestor(s.ctrl)
+	attestationData := &common.AttestedData{
+		Type: "join_token",
+		Data: []byte("foobar"),
+	}
+	fa := &nodeattestor.FetchAttestationDataResponse{
+		AttestedData: attestationData,
+		SpiffeId:     "spiffe://example.com/spire/agent/join_token/foobar",
+	}
+	s.nodeAttestor.EXPECT().FetchAttestationData(gomock.Any()).
+		Return(fa, nil)
+
+	svid, key, err := util.LoadSVIDFixture()
+	s.Require().NoError(err)
+
+	keyDer, err := x509.MarshalECPrivateKey(key)
+	s.Require().NoError(err)
+
+	s.keyManager = mock_keymanager.NewMockKeyManager(s.ctrl)
+	s.keyManager.EXPECT().FetchPrivateKey(gomock.Any()).Return(
+		&keymanager.FetchPrivateKeyResponse{PrivateKey: keyDer}, nil)
+	s.keyManager.EXPECT().GenerateKeyPair(gomock.Any()).Return(
+		&keymanager.GenerateKeyPairResponse{key.Y.Bytes(), keyDer}, nil)
+
+	s.catalog = mock_catalog.NewMockCatalog(s.ctrl)
+	s.catalog.EXPECT().NodeAttestors().
+		Return([]nodeattestor.NodeAttestor{s.nodeAttestor})
+	s.catalog.EXPECT().KeyManagers().
+		Return([]keymanager.KeyManager{s.keyManager})
+
+	s.nodeClient = mock_node.NewMockNodeClient(s.ctrl)
+	s.nodeClient.EXPECT().FetchBaseSVID(gomock.Any(), gomock.Any()).
+		Return(&node.FetchBaseSVIDResponse{&node.SvidUpdate{
+			Svids: map[string]*node.Svid{
+				"spiffe://example.com/spire/agent/join_token/foobar": &node.Svid{
+					svid.Raw,
+					300,
+				}},
+		}}, nil)
+	log, _ := test.NewNullLogger()
+	tempDir, err := ioutil.TempDir(os.TempDir(), "spire-test")
+	s.Require().NoError(err)
+
+	s.config = &Config{
+		Catalog: s.catalog,
+		DataDir: tempDir,
+		Log:     log,
+		TrustDomain: url.URL{
+			Scheme: "spiffe",
+			Host:   "example.com",
+		},
+		NodeClient: s.nodeClient,
+	}
+
+	s.attestor = New(s.config)
+}
+
+func (s *AttestorTestSuite) TeardownTest() {
+	os.RemoveAll(s.config.DataDir)
+	s.ctrl.Finish()
+}
+
+func (s *AttestorTestSuite) TestAttestLoadFromDisk() {
+	err := os.Link(
+		"../../../test/fixture/certs/bundle.der",
```

Comment:
test utils package has a `ProjectRoot()` function that can be used here along with a `path.Join`
```go
path.Join(utils.ProjectRoot(), "test/fixture/certs/bundle.der"
```